tools/pipstrap.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+"""Uses pip to upgrade Python packaging tools to pinned versions."""
+from __future__ import absolute_import
+
+import os
+import shutil
+import tempfile
+
+import pip_install
```

Comment:
I am not super fan of having pipstrap.py depending on pip_install here, because these two files are scripts and not modules formally associated into a package.

I appreciate the old pipstrap to be a self contained script that can just be copied anywhere while still behaving correctly.

However some other scripts in `tools` depend on each other, and the imported functions will me removed eventually.

So I am not against it, I just wanted to balance it versus self contained scripts at the cost if some code duplication.

Reply:
Yeah I think there's a lot of refactoring that can/should be done here.

I think a lot of our `pip` tools should be rethought and I agree putting any scripts with dependencies on each other in a package is a good idea.

I personally prefer reusing the code here though and assuming you're OK with that which it sounds like you are, I'll just create an issue for us to think about refactoring these scripts in the future.

Reply:
Agree!

Reply:
The issue I said I'd create is https://github.com/certbot/certbot/issues/8421.